integration_tests/test_employment.py

```diff
@@ -73,3 +75,153 @@ def test_movers_change_employment(populations):
+        ending_personal_income_propensity.index
+    )
+
+    assert (
+        starting_personal_income_propensity.loc[present_sims_idx]
+        == ending_personal_income_propensity.loc[present_sims_idx]
+    ).all()
+
+
+def test_income_updates(tracked_live_populations):
+    for time_step in range(max(TIME_STEPS_TO_TEST)):
+        (
+            employer_ids_before,
+            income_before,
+            employer_ids_after,
+            income_after,
+        ) = get_consecutive_time_step_employer_ids_and_test_column_for_comparable_sims(
+            tracked_live_populations, time_step, "income"
+        )
+
+        # Find those that change employers
+        changed_jobs_idx = employer_ids_before.index[
+            employer_ids_before != employer_ids_after
+        ]
+
+        assert (
+            income_before.loc[changed_jobs_idx] != income_after.loc[changed_jobs_idx]
+        ).all()
+
+
+def test_no_income_uupdate(tracked_live_populations):
+    for time_step in range(max(TIME_STEPS_TO_TEST)):
+        (
+            employer_ids_before,
+            income_before,
+            employer_ids_after,
+            income_after,
+        ) = get_consecutive_time_step_employer_ids_and_test_column_for_comparable_sims(
+            tracked_live_populations, time_step, "income"
+        )
+
+        # Find those that change employers
+        static_jobs_idx = employer_ids_before.index[employer_ids_before == employer_ids_after]
+
+        assert (income_before.loc[static_jobs_idx] == income_after.loc[static_jobs_idx]).all()
+
+
+def test_income_updates_when_age_changes(tracked_live_populations):
+    # This test checks whether simulants who have a birthdays and therefore will have a new income distribution
+    # Note: The other tests for income implementation are contigent on comparing simulants who are the same age and
+    #  therefore ignore this edge case.
+    for time_step in range(max(TIME_STEPS_TO_TEST)):
+        pop_before = tracked_live_populations[time_step]
+        pop_after = tracked_live_populations[time_step]
+        common_sims_idx = pop_before.index.intersection(pop_after.index)
+        age_before = np.floor(pop_before.loc[common_sims_idx, "age"])
+        age_after = np.floor(pop_after.loc[common_sims_idx, "age"])
+        birthday_sims_idx = age_before.index[age_before != age_after]
+        common_sims_idx = common_sims_idx.intersection(birthday_sims_idx)
+
+        # Get sims with same jobs but exclude unemployed
+        employer_ids_before = pop_before.loc[common_sims_idx, "employer_id"]
+        employer_ids_after = pop_after.loc[common_sims_idx, "employer_id"]
+        same_jobs_idx = employer_ids_before.index[employer_ids_before == employer_ids_after]
+
+        assert (
+            pop_before.loc[same_jobs_idx, "income"] != pop_after.loc[same_jobs_idx, "income"]
+        ).all()
+
+
+def get_consecutive_time_step_employer_ids_and_test_column_for_comparable_sims(
```

Comment:
I think this probably needs to be merged with #135 -- whether by you or me (depends who merges to main first, I guess).

Reply:
@albrja Specifically, as discussed this morning, this should make use of the `simulants_on_adjacent_timesteps` fixture, which is now merged into main.